### PR TITLE
mrc-1933: Add table data

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds the table data to mintr. 

It needs a bit of work in the upstream data itself, and in validating the the data conforms to the expected schema, so is draft until that's sorted out

### CODE REVIEW

**Review by hillalex — COMMENTED** [review]
This all looks good pending the data transformations that need to happen at the point of request once we are clear on what these are!

File: R/import.R [hunk 1]
```diff
@@ -55,8 +64,58 @@ mintr_db_process <- function(path) {
...
+
+  ## At this point casesAverted is really over 3 years and is cases
+  ## averted per person. This number can be greater than one as a
+  ## person can have more than one case per year. We remove the year
+  ## effect first:
+  table$casesAverted <- table$casesAverted / 3
+
+  ## Then compute the "per 1000" case before the uncertainty calculation:
+  table$casesAvertedPer1000 <- table$casesAverted * 1000
+
```
Comment (hillalex) [inline R/import.R:94]: Can we round both of the above to the nearest whole number? And also have:

```
  # We want these as percentages, rounded to the nearest d.p.
  table$reductionInCases <- round(table$reductionInCases * 100, 1)
  table$reductionInPrevalence <- round(table$reductionInPrevalence * 100, 1)
```

**Review by hillalex — COMMENTED** [review]
Couple of superficial edits to presentation of the numbers.

This is going to require some serious debugging :cry: The data is still completey different to the original app and there are still different series for `netType` `std` vs `pbo`.

For example, see that there are 2 data series with different values here coming back from mintr, and that neither of those series matches the one in the shiny prototype:

Screenshot from 2020-11-25 14-41-32

Screenshot from 2020-11-25 14-41-25

File: R/import.R (hunk 1 above)
Comment (richfitz) [reply R/import.R:94]: added this now

Comment (hillalex) [reply R/import.R:94]: Brill, thanks. Can we also round casesAverted and casesAvertedPer1000 to the nearest whole number?

File: R/db.R [hunk 2]
```diff
@@ -48,6 +53,13 @@ mintr_db <- R6::R6Class(
 
     get_cost_docs = function() {
       private$docs$cost
+    },
+
+    get_table = function(options) {
+      key <- self$get_index(options)
+      ret <- unserialize(private$db$get(sprintf("table:%s", key)))
+      ret$casesAverted <- round(ret$casesAverted * options$population, 1)
```
Comment (hillalex) [inline R/db.R:61]: can we make this just `round(ret$casesAverted * options$population)`?

File: R/import.R [hunk 3]
```diff
@@ -55,8 +64,62 @@ mintr_db_process <- function(path) {
...
+  table <- rename(table, unname(tr), names(tr))
+
+  ## At this point casesAverted is really over 3 years and is cases
+  ## averted per person. This number can be greater than one as a
+  ## person can have more than one case per year. We remove the year
+  ## effect first:
+  table$casesAverted <- table$casesAverted / 3
+
+  ## Then compute the "per 1000" case before the uncertainty calculation:
+  table$casesAvertedPer1000 <- round(table$casesAverted * 1000, 1)
```
Comment (hillalex) [inline R/import.R:93]: similarly `round(table$casesAverted * 1000)`

File: tests/testthat/test-db.R [hunk 4]
```diff
@@ -28,31 +30,36 @@ test_that("Can create db", {
...
-  expect_identical(
-    db$get_prevalence(c(options, population = 1000, metabolic = "yes")),
-    db$get_prevalence(options))
+                  sprayInput = "0%",
+                  metabolic = "yes",
+                  population = 1000)
+  d <- db$get_table(options)
+  expect_s3_class(d, "data.frame")
+  expect_equal(nrow(d), 114)
+  ## TODO: this will be fixed once we have the real data
```
Comment (hillalex) [inline tests/testthat/test-db.R:47]: this comment now obsolete?

**Review by hillalex — COMMENTED** [review]
Looks good! Just a couple of places where the rounding can be to the nearest whole number

**Review by hillalex — APPROVED** [review]

**Pull request merged**